purchase_receipt_progress/models/purchase_order.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrderReceivedByCategory(models.Model):
+    _name = "purchase.order.received.by.category"
+    _description = "Purchase Order Received by Category"
+
+    order_id = fields.Many2one("purchase.order")
+    category_id = fields.Many2one("product.category")
+    product_qty = fields.Float()
+    qty_received = fields.Float()
+    to_receive = fields.Float()
```

Comment:
Since we're handling quantities, we should also add the UoM management

Reply:
@SilvioC2C @gurneyalex 
Hmmm.. I don't know !
Here, the objective asked is to sum the quantities by category.
If we need to manage the UoM, we need to to group by category then by UoM because in the same product category maybe we'll have two (or more) different units..